src/Program.cs

```diff
@@ -1,245 +1,263 @@
-            DumpResponse("Re-generated Azure Search secondary admin key", response);
+            Console.WriteLine("----------------------------------------------------");
+            Console.WriteLine("List all search services in the subscription by resource group");
+            Console.WriteLine("----------------------------------------------------");
 
-            // Create a new query API key
-            response = ExecuteArmRequest(HttpMethod.Post, "resourcegroups/Default-Web-WestUS/providers/Microsoft.Search/searchServices/" + name + "/createQueryKey/myQueryKey?api-version=2014-07-31-Preview");
-            DumpResponse("Created new Azure Search service query key", response);
-
-            // Retrieve query API keys
-            response = ExecuteArmRequest(HttpMethod.Get, "resourcegroups/Default-Web-WestUS/providers/Microsoft.Search/searchServices/" + name + "/listQueryKeys?api-version=2014-07-31-Preview");
-            DumpResponse("List Azure Search service query keys", response);
-
-            // Delete a query API key (pick the first one)
-            string key = JsonConvert.DeserializeObject<dynamic>(response.Content.ReadAsStringAsync().Result).value[0].key;
-            response = ExecuteArmRequest(HttpMethod.Delete, "resourcegroups/Default-Web-WestUS/providers/Microsoft.Search/searchServices/" + name + "/deleteQueryKey/" + key + "?api-version=2014-07-31-Preview");
-            DumpResponse("Delete Azure Search service query key", response);
-
-            // Change service replica/partition count
-            // NOTE: this will fail unless you change the service creation code above to make it a "standard" service and wait until it's provisioned
-            response = ExecuteArmRequest(HttpMethod.Put,
-                                         "resourcegroups/Default-Web-WestUS/providers/Microsoft.Search/searchServices/" + name + "?api-version=2014-07-31-Preview",
-                                         new
-                                         {
-                                             type = "Microsoft.Search/searchServices",
-                                             location = "West US",
-                                             properties = new
-                                             {
-                                                 sku = new { name = "standard" },
-                                                 partitionCount = 1,
-                                                 replicaCount = 2
-                                             }
-                                         });
-            DumpResponse("Dynamically scale Azure Search service", response);
+            foreach (var group in groups)
+            {
+                var resources = await resourceClient.Resources.ListAsync("resourceGroup eq '" + group.Name + "' and resourceType eq 'Microsoft.Search/searchServices'");
```

Comment:
I think we ought to use the `SearchManagementClient` here to list the services in a resource group; Otherwise we're just demo-ing the ARM client.

Reply:
Can't list resources with the search management client.  Need the ARM client for that.  This is one of the steps in the existing sample converted over.  It's good to show when you need the ARM client vs ours for management, IMO.

Reply:
This code is enumerating search services specifically, not all resources. The search management client should be able to list services in either a resource group or subscription. So this is not actually a situation where you need the ARM client.